lib/IRGen/IRGenModule.cpp

```diff
@@ -1184,14 +1184,17 @@ void IRGenModule::emitAutolinkInfo() {
   }
 
   if (!IRGen.Opts.ForceLoadSymbolName.empty() &&
-      isFirstObjectFileInModule(*this)) {
+      (Triple.supportsCOMDAT() || isFirstObjectFileInModule(*this))) {
```

Comment:
Didn't we add this whole thing in the first place because we were using a COMMON global variable on Mach-O and that didn't work on Windows? Can't we go back to that on non-COMDAT platforms? (See #15647.)

Reply:
Yes, that is exactly what I did in #26635 (thanks to @rjmccall kicking my memory back into place!)